apps/reaction/tests/integration/api/mutations/surcharges/createSurcharge.test.js

```diff
@@ -122,3 +122,55 @@ test("an authorized user can create a surcharge", async () => {
   expect(result.createSurcharge.surcharge.attributes).toEqual(surchargeAttributes);
   expect(result.createSurcharge.surcharge.destination).toEqual(surchargeDestination);
 });
+
+test("an authorized user can create a surcharge with amount as 0", async () => {
+  await testApp.setLoggedInUser(mockAdminAccount);
+  let result;
+
+  try {
+    result = await createSurcharge({
+      createSurchargeInput: {
+        shopId: opaqueShopId,
+        surcharge: {
+          amount: 0,
+          messagesByLanguage: surchargeMessagesByLanguage,
+          type: "surcharge",
+          attributes: surchargeAttributes,
+          destination: surchargeDestination
+        }
+      }
+    });
+  } catch (error) {
+    expect(error).toBeUndefined();
+  }
```

Comment:
I think this would be enough.

```suggestion
  const  result = await createSurcharge({
      createSurchargeInput: {
        shopId: opaqueShopId,
        surcharge: {
          amount: 0,
          messagesByLanguage: surchargeMessagesByLanguage,
          type: "surcharge",
          attributes: surchargeAttributes,
          destination: surchargeDestination
        }
      }
    });

```